rotkehlchen/db/dbhandler.py

```diff
@@ -949,6 +954,277 @@ def delete_adex_events_data(self) -> None:
         self.conn.commit()
         self.update_last_write()
 
+    def add_balancer_events(
+            self,
+            events: Sequence[BalancerEvent],
+    ) -> None:
+        query = (
+            """
+            INSERT INTO balancer_events (
+                tx_hash,
+                log_index,
+                address,
+                timestamp,
+                type,
+                pool_address,
+                lp_amount,
+                usd_value,
+                amount0,
+                amount1,
+                amount2,
+                amount3,
+                amount4,
+                amount5,
+                amount6,
+                amount7
+            )
+            VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)
+            """
+        )
+        cursor = self.conn.cursor()
+        for event in events:
+            event_tuple = event.to_db_tuple()
+            try:
+                cursor.execute(query, event_tuple)
+            except sqlcipher.IntegrityError:  # pylint: disable=no-member
+                self.msg_aggregator.add_warning(
+                    f'Tried to add a Balancer event that already exists in the DB. '
+                    f'Event data: {event_tuple}. Skipping event.',
+                )
+                continue
+
+        self.conn.commit()
+        self.update_last_write()
+
+    def add_balancer_pools(
+            self,
+            pools: Sequence[BalancerEventPool],
+    ) -> None:
+        query = (
+            """
+            INSERT INTO balancer_pools (
```

Comment:
oh my god :rofl: 

But yeah I expected this. Thankfully this will all go away with the inclusion of tokens in the DB.